src/components/Avatar/index.js

```diff
@@ -7,13 +7,50 @@ const sizes = {
   4: 'w4 h4',
 }
 
-export default ({name, url, size = 3}) => (
-  <div
-    className={`bg-gray dib br-100 ${sizes[size]}`}
-    style={{
-      background: `url(${url}) center center / cover no-repeat`,
-    }}
-    role='img'
-    aria-label={`Avatar for ${name}`}
-  />
-)
+export default class extends Component {
+
+  static defaultProps = {
+    size: 3,
+  }
+
+  state = {
+    hasLoaded: false,
+  }
+
+  handleLoad = () => {
+    this.setState({hasLoaded: true});
+  }
+
+  render() {
+    const {hasLoaded} = this.state
+    const {name, url, size} = this.props
+    const alt = `Avatar for ${name}`
+    const containerClassName = `bg-gray-secondary dib br-100 ${sizes[size]}`
+
+    if(hasLoaded) {
+      return (
+        <div
+          className={containerClassName}
+          style={{
+            background: `url(${url}) center center / cover no-repeat`,
+          }}
+          role='img'
+          aria-label={`Avatar for ${name}`}
+        />
+      )
+    }
+
+    return (
+      <div className={containerClassName}>
+        <img
+          onLoad={this.handleLoad}
```

Comment:
The native DOM event for when an image loads. There is also `onError`, but this shows the background shade regardless of if it is loading or has errored.